TNDO-2893 Fork redisdown and switch to using individual keys for values

superquadratic opened the pull request:
To use Redis as a LRU cache backend for the nx [level-task-runner](https://www.npmjs.com/package/@apployees-nx/level-task-runner), we currently use [redisdown](https://github.com/hmalphettes/redisdown). However, redisdown puts all values (build/compute artefacts in this case) into one Redis hash. As a result, when the cache is full, Redis will "evict" (i.e. remove) old keys until there is enough free space. But, since everything is in one hash, everything is deleted.

To fix this situation, this is a fork of redisdown (that I called redis-key-down) that puts the values into individual keys, so that cache eviction works properly. It also includes a fix for an issue with Buffer values in redisdown 0.1.13 (https://github.com/leanix/redis-key-down/commit/288d196cf12639b0760da5b5a07257a95d4d9321) and for passing down parameters that are needed in our CI workflow (https://github.com/leanix/redis-key-down/commit/bb7e246423c588eafc8873d39a033677bed45b41).

gul-leanix committed:
Fix usage of Buffers as values

gul-leanix committed:
Rename cleanKey() function

gul-leanix committed:
Store values as individual keys to allow partial cache eviction

gul-leanix committed:
Allow passing tls and password options down to redis

gul-leanix committed:
Delete TravisCI config

gul-leanix committed:
Rename package and update README

gul-leanix committed:
Don't try to delete obsolete ':h' key

aleics-lx reviewed (commented):
Great job! Simple but powerful changes! 🥇 👏 

I have just a small question regarding the commands used :)

superquadratic commented on index.js:
```diff
@@ -111,7 +111,7 @@ RedisDown.prototype._open = function (options, callback) {
 
 RedisDown.prototype._get = function (key, options, callback) {
-    this.db.hget(this.location + ':h', toBufferOrString(key), function (e, v) {
+    this.db.get(valueKey(this.location, key), function (e, v) {
```
Exactly! `hget` (and "friends") work on keys that contains hash values (key-value pairs like JavaScript objects), so they take both the name of the Redis key and the name of the key in the hash. We now basically combine these two keys into one (`redisKey + '$' + hashKey`) and use the "plain" `get` command(s).

And you're right, implemented like this, `Buffer` keys will stop working. I will have a look.

gul-leanix committed:
Serialize all keys and values using abstract-leveldown callbacks

`_serializeKey()` and `_serializeValue` are called automatically by
abstract-leveldown before parameters are handed over to `_get()`,
`_put()` etc.

gul-leanix committed:
Support `Buffer` keys as well

aleics-lx reviewed (approved):
Great fix! 🚀 👏
